docs/ref/templates/builtins.txt

```diff
@@ -953,6 +953,71 @@ output (as a string) inside a variable. This is useful if you want to use
     {% now "Y" as current_year %}
     {% blocktranslate %}Copyright {{ current_year }}{% endblocktranslate %}
 
+.. templatetag:: query_string
+
+``query_string``
+----------------
+
+.. versionadded:: 5.1
+
+Outputs the query string from the given :class:`~django.http.QueryDict`
+instance, or ``request.GET`` if not given and the
+``django.template.context_processors.request`` context processor is enabled.
```

Comment:
Can we have a test for the behavior when this context processor is not enabled and a custom `QueryDict` is not provided? I presume that it'll currently just crash... Perhaps we want to make it clearer that the context processor needs to be enabled?

Reply:
I've added it. I also realised we were missing a test for when using a custom `QueryDict` is supplied without the context processor, which should and does work, so I've added that as well.

I added a warning into the docs as well, though not sure about the wording.